src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -17,13 +17,15 @@ private class NodeSelectorState
             public readonly int[] FastestRecords;
             public int Fastest;
             public int SpeedTestMode;
+            public int CurrentNodeIndex;
 
             public NodeSelectorState(Topology topology)
             {
                 Topology = topology;
                 Nodes = topology.Nodes;
                 Failures = new int[topology.Nodes.Count];
                 FastestRecords = new int[topology.Nodes.Count];
+                CurrentNodeIndex = 0;
```

Comment:
let's call it something like `UnlikelyEveryoneFaultedChoiceIndex`

Reply:
Done